File: src/geometryService/cpp/stractNet/DoubleFrag.cpp
///////////////////////////////////////////////////////////
//  DoubleFrag.cpp
//  Implementation of the Class DoubleFrag
//  Created on:      06-Aug-2008 7:52:13 AM
//  Original author: david.h.loman
///////////////////////////////////////////////////////////

#include "DoubleFrag.h"
///////////////////////////////////////////////////////////////////////
/*
 *
 * xSTRUCTORS
 *
 */
///////////////////////////////////////////////////////////////////////

DoubleFrag::~DoubleFrag(){}
DoubleFrag::DoubleFrag(double db){
	MsgFrag("DoubleFrag", db, MsgFrag::mftDouble);
}
DoubleFrag::DoubleFrag(DataInputStream& in){
	MsgFrag("DoubleFrag", in, MsgFrag::mftDouble);
}

///////////////////////////////////////////////////////////////////////
/*
 *
 * METHODS
 *
 */
///////////////////////////////////////////////////////////////////////
void DoubleFrag::Deserialize(DataInputStream& in){
	try {
		// Read Double
		this->field = in.readDouble();
	} catch (IOException ioe) {
		SNRoot::err("IOException in Deserialize()", 0);
	}
	return;

}

void DoubleFrag::Serialize_Specific(DataOutputStream& out){
	try {
			// write in new data
			out.writeDouble(this->field);
		} catch (IOException ioe) {
			SNRoot::err("IOException in Serialize()", 0);
		}

		return;

}

File: src/iBME/superceded_GS/java/stractNet/src/stractNet/messaging/msgs/frags/DoubleFrag.java
package stractNet.messaging.msgs.frags;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;

public class DoubleFrag extends MsgFrag<Double> {
	public DoubleFrag(double db) {
		super("DoubleFrag", db, MsgFrag.mftDouble);
	}

	public DoubleFrag(DataInputStream in) {
		super("DoubleFrag", in, MsgFrag.mftDouble);
	}

	@Override
	public void Deserialize(DataInputStream in) {
		try {
			// Read Double
			this.field = in.readDouble();
		} catch (IOException ioe) {
			this.err("IOException in Deserialize()", 0);
		}
		return;
	}

	@Override
	protected void Serialize_Specific(DataOutputStream out) {
		try {
			// write in new data
			out.writeDouble(this.field);
		} catch (IOException ioe) {
			this.err("IOException in Serialize()", 0);
		}

		return;
	}

}
